src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -826,11 +826,11 @@ private void DetermineAndAddExecutablePathCommand(IList<LaunchCommand> commands,
             if (launchOptions.UnixPort.IsOSX())
             {
                 // Usually the first FD=txt in the output of lsof points to the executable.
-                absoluteExePath = string.Format(CultureInfo.InvariantCulture, "shell lsof -p {0} | awk '$4 == \"txt\" {{ print $9 }}'|awk 'NR==1 {{print $1}}'", _launchOptions.ProcessId);
+                absoluteExePath = string.Format(CultureInfo.InvariantCulture, "shell lsof -p {0} | awk '$4 == \"txt\" {{ print $9 }}'|awk 'NR==1 {{print $1}}'", _launchOptions.ProcessId.HasValue ? _launchOptions.ProcessId.Value : 0);
             }
             else if (launchOptions.UnixPort.IsLinux())
             {
-                absoluteExePath = string.Format(CultureInfo.InvariantCulture, @"shell readlink -f /proc/{0}/exe", _launchOptions.ProcessId);
+                absoluteExePath = string.Format(CultureInfo.InvariantCulture, @"shell readlink -f /proc/{0}/exe", _launchOptions.ProcessId.HasValue ? _launchOptions.ProcessId.Value : 0);
```

Comment:
>HasValue [](start = 139, length = 8)

This method is only supposed to be called if the process id has a value. So I would suggest asserting and using `.Value`